packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -0,0 +1,761 @@
+    this.onDestinationSelected,
+    this.elevation,
+    this.groupAlignment,
+    this.labelType,
+    this.unselectedLabelTextStyle,
+    this.selectedLabelTextStyle,
+    this.unselectedIconTheme,
+    this.selectedIconTheme,
+    this.preferredWidth = _railWidth,
+    this.extendedWidth = _extendedRailWidth,
+  }) :  assert(destinations != null && destinations.isNotEmpty),
+        assert(0 <= currentIndex && currentIndex < destinations.length),
+        assert(elevation == null || elevation > 0),
+        assert(preferredWidth == null || preferredWidth > 0),
+        assert(extendedWidth == null || extendedWidth > 0),
+        assert((preferredWidth == null || extendedWidth == null) || extendedWidth >= preferredWidth),
+        assert(extended != null),
+        assert(!extended || (labelType == null || labelType == NavigationRailLabelType.none));
+
+  /// Sets the color of the Container that holds all of the [NavigationRail]'s
+  /// contents.
+  final Color backgroundColor;
+
+  /// Indicates of the [NavigationRail] should be in the extended state.
+  ///
+  /// The rail will implicitly animate between the extended and normal state.
+  ///
+  /// If the rail is going to be in the extended state, then the [labelType]
+  /// should be set to [NavigationRailLabelType.none].
+  final bool extended;
+
+  /// The leading widget in the rail that is placed above the destinations.
+  ///
+  /// This is commonly a [FloatingActionButton], but may also be a non-button,
+  /// such as a logo.
+  final Widget leading;
+
+  /// The trailing widget in the rail that is placed below the destinations.
+  ///
+  /// This is commonly a list of additional options or destinations that is
+  /// usually only rendered when [extended] is true.
+  final Widget trailing;
+
+  /// Defines the appearance of the button items that are arrayed within the
+  /// navigation rail.
+  final List<NavigationRailDestination> destinations;
+
+  /// The index into [destinations] for the current active
+  /// [NavigationRailDestination].
+  final int currentIndex;
+
+  /// Called when one of the [destinations] is selected.
+  ///
+  /// The stateful widget that creates the navigation rail needs to keep
+  /// track of the index of the selected [NavigationRailDestination] and call
+  /// `setState` to rebuild the navigation rail with the new [currentIndex].
+  final ValueChanged<int> onDestinationSelected;
+
+  /// The elevation for the inner side of the rail.
+  ///
+  /// The shadow only shows on the inner side of the rail.
+  ///
+  /// In LTR configurations, the inner side is the right side, and in RTL
+  /// configurations, it is the left side.
+  final double elevation;
+
+  /// The alignment for the [NavigationRailDestination]s as they are positioned
+  /// within the [NavigationRail].
+  ///
+  /// Navigation rail destinations can be aligned as a group to the [top],
+  /// [bottom], or [center] of a layout.
+  final NavigationRailGroupAlignment groupAlignment;
```

Comment:
This should really be a value between -1 and 1, like the Alignment class's y property.

Reply:
Done and updated doc